matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/state/DefaultStateService.kt

```diff
@@ -123,64 +113,52 @@ internal class DefaultStateService @AssistedInject constructor(@Assisted private
                                 // Sort for the cleanup
                                 .sorted()
                 ).toContent(),
-                callback = callback,
                 stateKey = null
         )
     }
 
-    override fun updateHistoryReadability(readability: RoomHistoryVisibility, callback: MatrixCallback<Unit>): Cancelable {
-        return sendStateEvent(
+    override suspend fun updateHistoryReadability(readability: RoomHistoryVisibility) {
+        sendStateEvent(
                 eventType = EventType.STATE_ROOM_HISTORY_VISIBILITY,
                 body = mapOf("history_visibility" to readability),
-                callback = callback,
                 stateKey = null
         )
     }
 
-    override fun updateJoinRule(joinRules: RoomJoinRules?, guestAccess: GuestAccess?, callback: MatrixCallback<Unit>): Cancelable {
-        return taskExecutor.executorScope.launchToCallback(coroutineDispatchers.main, callback) {
+    override suspend fun updateJoinRule(joinRules: RoomJoinRules?, guestAccess: GuestAccess?) {
+        withContext(coroutineDispatchers.main) {
             if (joinRules != null) {
-                awaitCallback<Unit> {
-                    sendStateEvent(
-                            eventType = EventType.STATE_ROOM_JOIN_RULES,
-                            body = RoomJoinRulesContent(joinRules).toContent(),
-                            callback = it,
-                            stateKey = null
-                    )
-                }
+                sendStateEvent(
+                        eventType = EventType.STATE_ROOM_JOIN_RULES,
+                        body = RoomJoinRulesContent(joinRules).toContent(),
+                        stateKey = null
+                )
             }
             if (guestAccess != null) {
-                awaitCallback<Unit> {
-                    sendStateEvent(
-                            eventType = EventType.STATE_ROOM_GUEST_ACCESS,
-                            body = RoomGuestAccessContent(guestAccess).toContent(),
-                            callback = it,
-                            stateKey = null
-                    )
-                }
+                sendStateEvent(
+                        eventType = EventType.STATE_ROOM_GUEST_ACCESS,
+                        body = RoomGuestAccessContent(guestAccess).toContent(),
+                        stateKey = null
+                )
             }
         }
     }
 
-    override fun updateAvatar(avatarUri: Uri, fileName: String, callback: MatrixCallback<Unit>): Cancelable {
-        return taskExecutor.executorScope.launchToCallback(coroutineDispatchers.main, callback) {
+    override suspend fun updateAvatar(avatarUri: Uri, fileName: String) {
+        withContext(coroutineDispatchers.main) {
```

Comment:
and here

Reply:
same as https://github.com/vector-im/element-android/pull/2500/files/c7efd1feb99179585e865affa866d680a96fefe2#r538291540